optaweb-tsp-planner/pom.xml

```diff
@@ -54,12 +55,43 @@
     </dependency>
   </dependencies>
 
+  <profiles>
+    <profile>
+      <id>include-ui</id>
+      <activation>
+        <activeByDefault>true</activeByDefault>
+      </activation>
+      <dependencies>
+        <dependency>
+          <groupId>org.optaweb.tsp</groupId>
+          <artifactId>optaweb-tsp-ui</artifactId>
+          <version>${project.version}</version>
+        </dependency>
+      </dependencies>
+    </profile>
+  </profiles>
   <build>
     <plugins>
       <plugin>
         <groupId>org.springframework.boot</groupId>
         <artifactId>spring-boot-maven-plugin</artifactId>
       </plugin>
+      <plugin>
+        <artifactId>maven-dependency-plugin</artifactId>
+        <executions>
+          <execution>
+            <id>unpack-dependencies</id>
+            <phase>prepare-package</phase>
+            <goals>
+              <goal>unpack-dependencies</goal>
+            </goals>
+            <configuration>
+              <outputDirectory>${project.build.directory}/classes</outputDirectory>
+              <excludeTransitive>true</excludeTransitive>
+            </configuration>
+          </execution>
+        </executions>
```

Comment:
@danielefiungo This has an unpleasant side effect. It unpacks all of this project's dependencies (including Spring Boot starters, H2 database, optaplanner examples) producing a sort of fat JAR. Some resources get overridden (e.g logback.xml). Can you configure the plugin to only unpack the `optaweb-tsp-ui` dependency?

Reply:
I've created and removed a review by mistake, anyway ...
the configuration prevent a minor side effect related to the unpack of UI jar that can be addressed with a mvn clean,  is minor compared your suggested side effects